index.js

```diff
@@ -211,6 +211,56 @@ function componentVersionTsMacroCheck(context, node) {
   }
 }
 
+function componentActionAnnotationsCheck(context, node) {
+  const component = getComponentFromNode(node);
+
+  if (!component) return;
+  const { properties } = component;
+
+  const typeProp = findPropertyWithName("type", properties);
+  if (typeProp?.value?.value !== "action") return;
+
+  const annotationsProp = findPropertyWithName("annotations", properties);
+
+  // Error 1 - annotations missing entirely
+  if (!annotationsProp) {
+    context.report({
+      node: component,
+      message: "Action component is missing required 'annotations' object",
+    });
+    return;
+  }
+
+  // Error 2 - annotations is not an object expression
+  if (annotationsProp.value.type !== "ObjectExpression") {
+    context.report({
+      node: annotationsProp.value,
+      message: "Property 'annotations' must be an object expression",
+    });
+    return;
+  }
+
+  // Error 3 - required keys missing
+  const requiredKeys = [
+    "destructiveHint",
+    "openWorldHint",
+    "readOnlyHint",
+  ];
+
+  const annotationKeys = annotationsProp.value.properties.map(
+    (prop) => prop.key && (prop.key.name || prop.key.value)
```

Comment:
To fix missing trailing comma `eslint(`[`comma-dangle`](https://eslint.org/docs/rules/comma-dangle)`)`
```suggestion
    (prop) => prop.key && (prop.key.name || prop.key.value),
```